File: app/src/main/java/uk/co/ribot/androidboilerplate/ui/main/MainActivity.java
package uk.co.ribot.androidboilerplate.ui.main;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;

import uk.co.ribot.androidboilerplate.R;
import uk.co.ribot.androidboilerplate.data.SyncService;
import uk.co.ribot.androidboilerplate.ui.base.BaseActivity;
import uk.co.ribot.androidboilerplate.utils.ActivityUtil;

public class MainActivity extends BaseActivity {

    private static final String EXTRA_TRIGGER_SYNC_FLAG =
            "uk.co.ribot.androidboilerplate.ui.main.MainActivity.EXTRA_TRIGGER_SYNC_FLAG";

    /**
     * Return an Intent to start this Activity.
     * triggerDataSyncOnCreate allows disabling the background sync service onCreate. Should
     * only be set to false during testing.
     */
    public static Intent getStartIntent(Context context, boolean triggerDataSyncOnCreate) {
        Intent intent = new Intent(context, MainActivity.class);
        intent.putExtra(EXTRA_TRIGGER_SYNC_FLAG, triggerDataSyncOnCreate);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MainFragment mainFragment = (MainFragment) getSupportFragmentManager()
                .findFragmentById(R.id.contentFrame);
        if (mainFragment == null) {
            // Create the fragment
            mainFragment = MainFragment.newInstance();
            ActivityUtil.addFragmentToActivity(
                    getSupportFragmentManager(), mainFragment, R.id.contentFrame);
        }

        if (getIntent().getBooleanExtra(EXTRA_TRIGGER_SYNC_FLAG, true)) {
            startService(SyncService.getStartIntent(this));
        }

        if (!(ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
                == PackageManager.PERMISSION_GRANTED) ||
                !(ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                == PackageManager.PERMISSION_GRANTED)) {
            ActivityCompat.requestPermissions(this, new String[]{
                    Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, 1000);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

}

File: app/src/main/java/uk/co/ribot/androidboilerplate/ui/main/MainActivity.kt
package uk.co.ribot.androidboilerplate.ui.main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import uk.co.ribot.androidboilerplate.R
import uk.co.ribot.androidboilerplate.data.SyncService
import uk.co.ribot.androidboilerplate.ui.base.BaseActivity
import uk.co.ribot.androidboilerplate.utils.addFragmentToActivity

class MainActivity : BaseActivity() {

    companion object {
        private const val EXTRA_TRIGGER_SYNC_FLAG =
                "uk.co.ribot.androidboilerplate.ui.main.MainActivity.EXTRA_TRIGGER_SYNC_FLAG"

        /**
         * Return an Intent to start this Activity.
         * triggerDataSyncOnCreate allows disabling the background sync service onCreate. Should
         * only be set to false during testing.
         */
        fun getStartIntent(context: Context, triggerDataSyncOnCreate: Boolean): Intent {
            val intent = Intent(context, MainActivity::class.java)
            intent.putExtra(EXTRA_TRIGGER_SYNC_FLAG, triggerDataSyncOnCreate)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var mainFragment: MainFragment? = supportFragmentManager
                .findFragmentById(R.id.contentFrame) as? MainFragment
        mainFragment ?: let {
            // Create the fragment
            mainFragment = MainFragment.newInstance()
            addFragmentToActivity(supportFragmentManager, mainFragment!!, R.id.contentFrame)
        }

        if (intent.getBooleanExtra(EXTRA_TRIGGER_SYNC_FLAG, true)) {
            startService(SyncService.getStartIntent(this))
        }
    }

    override fun onDestroy() {
        super.onDestroy()
    }

}
